cpp/src/stream_compaction/drop_duplicates.cu

```diff
@@ -139,6 +139,55 @@ get_unique_ordered_indices(const cudf::table& key_columns,
   return std::make_pair(unique_indices, 
                         thrust::distance(unique_indices.begin(), result_end));
 }
+
+gdf_size_type unique_count(const cudf::table& key_columns,
+             const bool nulls_are_equal = true,
+             cudaStream_t stream=0)
+{
+  gdf_size_type ncols = key_columns.num_columns();
+  gdf_size_type nrows = key_columns.num_rows();
+
+  // sort only indices
+  rmm::device_vector<gdf_size_type> sorted_indices(nrows);
+  gdf_context context;
+  gdf_column sorted_indices_col;
+  CUDF_TRY(gdf_column_view(&sorted_indices_col, static_cast<void*>(sorted_indices.data().get()),
+        nullptr, nrows, GDF_INT32));
+  CUDF_TRY(gdf_order_by(key_columns.begin(),
+        nullptr,
+        key_columns.num_columns(),
+        &sorted_indices_col,
+        &context));
+
+  // count unique elements
+  auto sorted_index_iter = sorted_indices.begin();
+  auto exec = rmm::exec_policy(stream)->on(stream);
+  auto device_input_table = device_table::create(key_columns, stream);
+
+  bool nullable = device_input_table->has_nulls();
+  if(cudf::has_nulls(key_columns)) {
+    auto comp = row_equality_comparator<true>(*device_input_table,
+        nulls_are_equal);
+    return thrust::count_if(exec,
+              thrust::counting_iterator<gdf_size_type>(0),
+              thrust::counting_iterator<gdf_size_type>(nrows),
+              [first = sorted_index_iter, comp] 
+              __device__ (const gdf_size_type i) {
+              return (i == 0 || !comp(first[i], first[i-1]));
```

Comment:
`first` doesn't convey a lot of meaning. I suggest naming more along these lines. 
```suggestion
              return (i == 0 || !comp(sorted_order[i], sorted_order[i-1]));
```